command/agent/consul/connect.go

```diff
@@ -226,38 +233,59 @@ func connectMeshGateway(in structs.ConsulMeshGateway) api.MeshGatewayConfig {
 	return gw
 }
 
-func connectProxyConfig(cfg map[string]interface{}, port int, allocID string) map[string]interface{} {
+func connectProxyConfig(cfg map[string]interface{}, port int, info allocInfo) map[string]interface{} {
 	if cfg == nil {
 		cfg = make(map[string]interface{})
 	}
 	cfg["bind_address"] = "0.0.0.0"
 	cfg["bind_port"] = port
-	injectAllocID(cfg, allocID)
+
+	tags := map[string]string{
+		"nomad.group=":     info.Group,
+		"nomad.job=":       info.JobID,
+		"nomad.namespace=": info.Namespace,
+		"nomad.alloc_id=":  info.AllocID,
+	}
+	injectNomadInfo(cfg, tags)
 	return cfg
 }
 
-// injectAllocID merges allocID into cfg=>envoy_stats_tags
+// injectNomadInfo merges nomad information into cfg=>envoy_stats_tags
 //
 // cfg must not be nil
-func injectAllocID(cfg map[string]interface{}, allocID string) {
+func injectNomadInfo(cfg map[string]interface{}, defaultTags map[string]string) {
 	const key = "envoy_stats_tags"
-	const prefix = "nomad.alloc_id="
-	pair := prefix + allocID
 	tags, exists := cfg[key]
 	if !exists {
-		cfg[key] = []string{pair}
+		var tags []string
+		for k, value := range defaultTags {
+			if value == "" {
+				continue
+			}
+			pair := k + value
+			tags = append(tags, pair)
+		}
+		if tags != nil {
+			cfg[key] = tags
+		}
 		return
 	}
-
 	switch v := tags.(type) {
```

Comment:
I realize this is existing code but let's rename this variable while we're here (something like `existingTags` or `cfgTags`?). "v" suggest it's a "value", which is really confusing in the context where we've got key, value pairs being iterated over.